src/store/tests/tests.effects.ts

```diff
@@ -0,0 +1,449 @@
+import { Injectable } from '@angular/core';
+import {
+  Actions,
+  createEffect,
+  Effect,
+  ofType,
+} from '@ngrx/effects';
+import { from, of, interval } from 'rxjs';
+import { Store, select, Action } from '@ngrx/store';
+import {
+  find, startsWith, omit, has,
+} from 'lodash';
+import { HttpResponse, HttpErrorResponse } from '@angular/common/http';
+import { TestSlot } from '@dvsa/mes-journal-schema';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import {
+  Examiner,
+  TestSlotAttributes,
+  ConductedLanguage,
+  CategoryCode,
+} from '@dvsa/mes-test-schema/categories/common';
+
+import {
+  switchMap, catchError, filter, map, withLatestFrom, concatMap,
+} from 'rxjs/operators';
+import * as testActions from './tests.actions';
+import * as testStatusActions from './test-status/test-status.actions';
+import {
+  PopulateApplicationReference,
+} from './journal-data/common/application-reference/application-reference.actions';
+import { PopulateCandidateDetails } from './journal-data/common/candidate/candidate.actions';
+import { testReportPracticeModeSlot } from './__mocks__/tests.mock';
+import { getTests } from './tests.reducer';
+import {
+  getCurrentTest,
+  isPracticeMode,
+  getCurrentTestSlotId,
+  getCurrentTestStatus,
+} from './tests.selector';
+import { TestStatus } from './test-status/test-status.model';
+import { TestsModel } from './tests.model';
+import { getJournalState } from '../journal/journal.reducer';
+import { getSlotsOnSelectedDate } from '../journal/journal.selector';
+import { PopulateExaminer } from './journal-data/common/examiner/examiner.actions';
+import { PopulateTestSlotAttributes } from './journal-data/common/test-slot-attributes/test-slot-attributes.actions';
+import { extractTestSlotAttributes } from './journal-data/common/test-slot-attributes/test-slot-attributes.selector';
+import { PopulateTestCentre } from './journal-data/common/test-centre/test-centre.actions';
+import { extractTestCentre } from './journal-data/common/test-centre/test-centre.selector';
+import { PopulateTestCategory } from './category/category.actions';
+import { PopulateTestSchemaVersion } from './schema-version/schema-version.actions';
+import { SetExaminerBooked } from './examiner-booked/examiner-booked.actions';
+import { SetExaminerConducted } from './examiner-conducted/examiner-conducted.actions';
+import { SetExaminerKeyed } from './examiner-keyed/examiner-keyed.actions';
+import { MarkAsRekey } from './rekey/rekey.actions';
+// import { getRekeySearchState, RekeySearchModel } from '../../pages/rekey-search/rekey-search.reducer';
```

Comment:
think a lot of these commented imports can be deleted

Reply:
Will do